object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/ExpressionBuilder.java

```diff
@@ -90,7 +90,12 @@ public void addIn(PropertyPath<?> propertyPath, List<Object> values) {
          ComparisonExpr booleanExpr = new ComparisonExpr(valueExpr, new ConstantValueExpr(typedValue), ComparisonExpr.Type.EQUAL);
          children.add(booleanExpr);
       }
-      push(new OrExpr(children));
+      if (children.size() == 1) {
+         // simplify INs with just one clause by removing the wrapper boolean expression
+         push(children.get(0));
+      } else {
+         push(new OrExpr(children));
```

Comment:
@gustavonalle This looks like an optimization, not a fix... is there a problem with `OrExpr` when it has a single child, and shouldn't we fix that as well?

Reply:
@danberindei I can certify this change fixes the reproducer, I'll let @anistor comment on other possible cases